esmerald/injector.py

```diff
@@ -9,6 +9,28 @@
     from esmerald.typing import AnyCallable
 
 
+class Factory:
+    def __init__(self, provides: "AnyCallable", *args: Any) -> None:
+        self.provides = provides
+        self.__args: Tuple[Any, ...] = ()
```

Comment:
shouldn't this be:

```python
self.__args: Tuple[Any, ...] = args
```
?

Reply:
This can be done directly, but personally I prefer to use the getter and setter way of dealing with internal objects:
```python
        self.__args: Tuple[Any, ...] = ()
        self.set_args(*args)
```